xbmc/android/activity/XBMCApp.cpp

```diff
@@ -101,6 +102,7 @@ CXBMCApp::CXBMCApp(ANativeActivity* nativeActivity)
   : CJNIApplicationMainActivity(nativeActivity)
   , CJNIBroadcastReceiver("org/xbmc/kodi/XBMCBroadcastReceiver")
 {
+  m_xbmcappinstance = this;
```

Comment:
Good question. No obvious reasons, so historical, I guess.
Making it a singleton means a hefty refactoring, though...